src/components/molecules/tabs/request/response/VisualEventLog.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+	ListItemDecorator,
+	ListItemContent,
+	Typography,
+	IconButton,
+	ListItem,
+	Divider,
+	List,
+	Box,
+	Stack,
+	ListItemButton,
+} from '@mui/joy';
+import { useContext, useState } from 'react';
+import { AuditLog, TransformedAuditLog, auditLogManager } from '../../../../../managers/AuditLogManager';
+import { ApplicationDataContext, TabsContext } from '../../../../../managers/GlobalContextManager';
+import { tabsManager } from '../../../../../managers/TabsManager';
+import { iconFromTabType } from '../../../../../types/application-data/application-data';
+import { camelCaseToTitle, formatMilliseconds } from '../../../../../utils/string';
+import { SprocketTooltip } from '../../../../atoms/SprocketTooltip';
+import TimerIcon from '@mui/icons-material/Timer';
+import AnchorIcon from '@mui/icons-material/Anchor';
+import BadgeIcon from '@mui/icons-material/Badge';
+import LaunchIcon from '@mui/icons-material/Launch';
+import SendIcon from '@mui/icons-material/Send';
+import CodeIcon from '@mui/icons-material/Code';
+import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import WhereToVoteIcon from '@mui/icons-material/WhereToVote';
+import { CollapseExpandButton } from '../../../../atoms/buttons/CollapseExpandButton';
+
+const eventStrIconsMap = {
+	Service: (
+		<>
+			{iconFromTabType.service}
+			<CodeIcon />
+		</>
+	),
+	Endpoint: (
+		<>
+			{iconFromTabType.endpoint}
+			<CodeIcon />
+		</>
+	),
+	Request: (
+		<>
+			{iconFromTabType.request}
+			<CodeIcon />
+		</>
+	),
+	request: (
+		<>
+			{iconFromTabType.request}
+			<SendIcon />
+		</>
+	),
+	root: (
+		<>
+			<AnchorIcon />
+		</>
+	),
+};
+
+export function VisualEventLog(props: { auditLog: AuditLog; requestId: string }) {
+	const data = useContext(ApplicationDataContext);
+	const tabsContext = useContext(TabsContext);
+	function VisualEventLogInner(props: { transformedLog: TransformedAuditLog; requestId: string }) {
```

Comment:
If at all possible, I think it would be best if this was pulled out into a separate component., especially since it has additional nested elements.